tools/tsconfig.lib.json

```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "ES2015",
+    "outDir": "../dist/out-tsc",
```

Comment:
I don't think we ever made a decision about whether to put the output in a single `dist` folder at the root. Unless it's necessary for nx to have it at the root for some reason, could it stay inside the `tools` folder for now?

Reply:
> Unless it's necessary for nx to have it at the root for some reason, could it stay inside the tools folder for now

- right, this is default nx boilerplate. In general every build output is stored within root `dist/` -> `/dist/<package-name>/....`.
  -   a) this is required for TS when path aliases are used ( currently we "hack it" by removing aliases during lage builds)
  -   b) I mentioned reasoning in [this RFC](https://github.com/microsoft/fluentui/pull/17298#discussion_r593089601)
- in reality within /tools, this config file is solely for proper type checking per environment ( in this case, `lib` -> implementation code )